src/Orleans.Core/Configuration/OptionLogger/OptionFormatterExtensionMethods.cs

```diff
@@ -42,35 +42,41 @@ public static class OptionConfigureExtensionMethods
         {
             var registration = services.FirstOrDefault(service => service.ServiceType == typeof(IOptionFormatter<TOptions>));
             if (registration == null)
-                services.AddSingleton<IOptionFormatter<TOptions>, TOptionFormatter>()
-                        .AddFromExisting<IOptionFormatter, IOptionFormatter<TOptions>>();
+                services.ConfigureFormatter<TOptions, TOptionFormatter>();
             return services;
         }
 
         /// <summary>
         /// Configure option formatter resolver for named option TOptions
         /// </summary>
-        public static IServiceCollection ConfigureFormatter<TOptions, TOptionFormatterResolver>(this IServiceCollection services, string name)
+        public static IServiceCollection ConfigureFormatterResolver<TOptions, TOptionFormatterResolver>(this IServiceCollection services)
             where TOptions : class
             where TOptionFormatterResolver : class, IOptionFormatterResolver<TOptions>
         {
-            services.AddSingleton<IOptionFormatterResolver<TOptions>, TOptionFormatterResolver>()
-                    .AddSingleton<IOptionFormatter>(sp => sp.GetService<IOptionFormatterResolver<TOptions>>().Resolve(name));
-            return services;
+            return services.AddSingleton<IOptionFormatterResolver<TOptions>, TOptionFormatterResolver>();
         }
 
         /// <summary>
         /// Configure option formatter resolver for named option TOptions, if none is configured
         /// </summary>
-        public static IServiceCollection TryConfigureFormatter<TOptions, TOptionFormatterResolver>(this IServiceCollection services, string name)
+        public static IServiceCollection TryConfigureFormatterResolver<TOptions, TOptionFormatterResolver>(this IServiceCollection services)
             where TOptions : class
             where TOptionFormatterResolver : class, IOptionFormatterResolver<TOptions>
         {
             var registration = services.FirstOrDefault(service => service.ServiceType == typeof(IOptionFormatterResolver<TOptions>));
             if (registration == null)
-                services.AddSingleton<IOptionFormatterResolver<TOptions>, TOptionFormatterResolver>();
-            services.AddSingleton<IOptionFormatter>(sp => sp.GetService<IOptionFormatterResolver<TOptions>>().Resolve(name));
+                return services.ConfigureFormatterResolver<TOptions, TOptionFormatterResolver>();
             return services;
         }
+
+        /// <summary>
+        /// Configure a named option to be logged
+        /// </summary>
+        public static IServiceCollection ConfigureNamedOptionForLogging<TOptions>(this IServiceCollection services, string name)
```

Comment:
I'm not entirely sure what's the pattern here to use this one. So I think the idea is this function can only be called once. But how do we enforce it ?

Reply:
nvm.  I think I understand it

Reply:
We don't.  If users call it twice, the get information logged twice.  :/ User error.

Reply:
I thought I understood, But I was wrong.
So consider this scenario. In one of our in house provider, `AddAzureTableGrainStorage` for example, we would call `ConfigureNamedOptionForLogging` in the method to configure formatter by default. If user want to configure a different one, they would call `ConfigureNamedOptionForLogging` again. So really user cannot avoid calling this twice.

Reply:
Why would a user call ConfigureNamedOptionForLogging?  There is never a need unless they've added a new named option.  If they want to change the formatter for a type of named option they'd call ConfiguredConfigureFormatterResolver.

Reply:
Sorry I asked mainly because the use pattern of `ConfigureNamedOptionForLogging` isn't clear to me. But we cleared that out in person.... I wasn't trying to push back or anything, because after all this pr improves the feature. We can merge this after tests passing

Reply:
all good.  Pattern still has some holes we can/should plug.